tests/solana.rs

```diff
@@ -1017,40 +1017,37 @@ fn translate(
     access_type: AccessType,
     vm_addr: u64,
     len: u64,
-) -> ProgramResult {
-    memory_mapping.map(access_type, vm_addr, len, 0)
+) -> Result<u64, Error> {
+    memory_mapping.map(access_type, vm_addr, len, 0).into()
 }
 
 fn translate_type_inner<'a, T>(
     memory_mapping: &MemoryMapping,
     access_type: AccessType,
     vm_addr: u64,
-) -> Result<&'a mut T, EbpfError> {
-    unsafe {
-        match translate(memory_mapping, access_type, vm_addr, size_of::<T>() as u64) {
-            ProgramResult::Ok(value) => Ok(&mut *(value as *mut T)),
-            ProgramResult::Err(e) => Err(e),
-        }
-    }
+) -> Result<&'a mut T, Error> {
+    let host_addr = translate(memory_mapping, access_type, vm_addr, size_of::<T>() as u64)?;
+
+    Ok(unsafe { &mut *(host_addr as *mut T) })
```

Comment:
I guess in the test it's not terribly important. But usually `unsafe {}` block come with a comment explaining their safety assumptions.

Reply:
I've added a comment